NetTopologySuite.IO/NetTopologySuite.IO/IO/Streams/FileStreamProvider.cs

```diff
@@ -19,10 +21,10 @@ public FileStreamProvider(string kind, string path, bool validatePath = false)
             if (path == null)
                 throw new ArgumentNullException("path");
 
-#if NET40
+#if HAS_SYSTEM_STRING_ISNULLORWHITESPACE
             if (string.IsNullOrWhiteSpace(path))
 #else
-            if (string.IsNullOrEmpty(path))
+            if (string.IsNullOrEmpty(path) || path.All(Char.IsWhiteSpace))

```

Comment:
`StringEx.IsNullOrWhiteSpaceNTS` uses a regex, but this should not be awful: in almost all cases, I expect it to terminate after reading exactly one character, and this all happens very close to filesystem operations that drown out this cost, so I'm going to leave it alone instead of either try to reference that method (it's internal in a different assembly) or copy the regex here.